server/src/main/java/org/opensearch/index/fielddata/IndexNumericFieldData.java

```diff
@@ -133,8 +134,16 @@ public final SortField sortField(
             : SortedNumericSelector.Type.MIN;
         SortField sortField = new SortedNumericSortField(getFieldName(), getNumericType().sortFieldType, reverse, selectorType);
         sortField.setMissingValue(source.missingObject(missingValue, reverse));
+        
+        // For long, doubles & dates, we can use this numeric sort optimiser.
+        boolean enablePointsOptimization = Arrays.asList(NumericType.DATE,
```

Comment:
I still tend to prefer the suggestion to move this into a property within NumericType, but if we go with this approach it would seem preferable to create a static set (or use a switch statement) instead of building a new list instance every time.

Reply:
Done. I added to NumericType. Thanks for suggstion.

Reply:
-1. I removed this...it's overly cryptic. We need to keep this selective optimization close to the source, not tucked away in the ctor. We can beautify it in a follow up PR that adds the optimization for all numeric type (which will require logic to handle types when merging results from different indices.) I believe @reta has been looking at this for a while. In the meantime lets keep this simple.

Reply:
Note also that I included this in the code comment around the block to keep this change and todo very explicit.